src/utils/hooks/useGetResponsesAndElementState.tsx

```diff
@@ -0,0 +1,127 @@
+import { useState, useEffect } from 'react'
+import { useGetElementsAndResponsesQuery, GetElementsAndResponsesQuery } from '../generated/graphql'
+import {
+  ResponsesByCode,
+  ResponsesFullByCode,
+  TemplateElementState,
+  EvaluatedElement,
+} from '../types'
+import evaluateExpression from '@openmsupply/expression-evaluator'
+
+const useGetResponsesAndElementState = (props: { serialNumber: string }) => {
+  const { serialNumber } = props
+  const [responsesByCode, setResponsesByCode] = useState({})
+  const [responsesFullByCode, setResponsesFullByCode] = useState({})
+  const [elementsExpressions, setElementsExpressions] = useState([{}])
+  const [elementsState, setElementsState] = useState({})
+  const [error, setError] = useState('')
+  const [loading, setLoading] = useState(true)
+  const { data, loading: apolloLoading, error: apolloError } = useGetElementsAndResponsesQuery({
+    variables: { serial: serialNumber },
+  })
+
+  useEffect(() => {
+    const error = checkForApplicationErrors(data)
+
+    if (error) {
+      setError(error)
+      setLoading(false)
+      return
+    }
+
+    if (apolloError) return
+
+    const applicationResponses = data?.applicationBySerial?.applicationResponses?.nodes
+
+    const templateSections = data?.applicationBySerial?.template?.templateSections?.nodes
+
+    const templateElements: TemplateElementState[] = []
+
+    templateSections?.forEach((sectionNode) => {
+      sectionNode?.templateElementsBySectionId?.nodes.forEach((element) => {
+        if (element) templateElements.push(element as TemplateElementState) // No idea why ...[spread] doesn't work here.
```

Comment:
```suggestion
templateSections.forEach((sectionNode) => {
      const elementsInSection = sectionNode.templateElementsBySectionId?.nodes as TemplateElement[]
      elementsInSection.forEach((element) => {
         templateElements.push(element as TemplateElementState) // No idea why ...[spread] doesn't work here.
```

Reply:
> No idea why ...[spread] doesn't work here.

I think the problem is that the types aren't the same. The structure of `TemplateElementState` isn't considering `undefined` which is a possible return from the GraphQL results for each `TemplateElement`.

Reply:
This suggestion doesn't work. `sectionNode` is underlined, with the error `Object is possibly 'null'`.  🤷‍♂️

Reply:
Nevermind, it works after I apply the suggestion further up. 👍 thx.